src/utils/openmc/settings.py

```diff
@@ -0,0 +1,1247 @@
+
+        if not is_integer(inactive):
+            msg = 'Unable to set inactive batches to a non-integer ' \
+                  'value {0}'.format(inactive)
+            raise ValueError(msg)
+
+        if inactive <= 0:
+            msg = 'Unable to set inactive batches to a negative ' \
+                  'value {0}'.format(inactive)
+            raise ValueError(msg)
+
+        self._inactive = inactive
+
+
+    def set_particles(self, particles):
+
+        if not is_integer(particles):
+            msg = 'Unable to set particles to a non-integer ' \
+                  'value {0}'.format(particles)
+            raise ValueError(msg)
+
+        if particles <= 0:
+            msg = 'Unable to set particles to a negative ' \
+                  'value {0}'.format(particles)
+            raise ValueError(msg)
+
+        self._particles = particles
+
+
+    def set_source_file(self, source_file):
+
+        if not is_string(source_file):
+            msg = 'Unable to set source file to a non-string ' \
+                  'value {0}'.format(source_file)
+            raise ValueError(msg)
+
+        self._source_file = source_file
+
+
+    def set_source_space(self, type, params):
+
+        if not is_string(type):
+            msg = 'Unable to set source space type to a non-string ' \
+                  'value {0}'.format(type)
+            raise ValueError(msg)
+
+        elif not type in ['box', 'point']:
+            msg = 'Unable to set source space type to {0} since it is not ' \
+                  'box or point'.format(type)
+            raise ValueError(msg)
+
+        elif not isinstance(params, (tuple, list, np.ndarray)):
+            msg = 'Unable to set source space parameters to {0} since it is ' \
+                  'not a Python tuple, list or NumPy array'.format(params)
+            raise ValueError(msg)
+
+        elif len(params) != 6:
+            msg = 'Unable to set source space parameters to {0} since it ' \
+                  'does not contain 6 values'.format(params)
+            raise ValueError(msg)
+
+        for param in params:
+
+            if not is_integer(param) and not is_float(param):
+                msg = 'Unable to set source space parameters to {0} since it ' \
+                      'is not an integer or floating point value'.format(param)
+                raise ValueError(msg)
+
+        self._source_space_type = type
+        self._source_space_params = params
+
+
+    def set_source_angle(self, type, params=[]):
+
+        if not is_string(type):
+            msg = 'Unable to set source angle type to a non-string ' \
+                  'value {0}'.format(type)
+            raise ValueError(msg)
+
+        elif not type in ['isotropic', 'monodirectional']:
+            msg = 'Unable to set source angle type to {0} since it is not ' \
+                  'isotropic or monodirectional'.format(type)
+            raise ValueError(msg)
+
+        elif not isinstance(params, (tuple, list, np.ndarray)):
```

Comment:
That wouldn't cover numpy.ndarray though. Are you suggesting `(collections.Sequence, np.ndarray)`?